modAionImpl/test/org/aion/zero/impl/cli/CliTest.java

```diff
@@ -0,0 +1,1167 @@
+        // test 1: -d -n last
+        ECKey key = ECKeyFac.inst().create();
+        String pKey = Hex.toHexString(key.getPrivKeyBytes());
+
+        assertEquals(
+                EXIT,
+                mockCli.call(
+                        new String[] {option, pKey, "-d", dataDirectory, "-n", "mastery"}, cfg));
+        assertThat(Keystore.list().length).isEqualTo(1);
+
+        // test 2: known key
+        assertEquals(
+                ERROR,
+                mockCli.call(
+                        new String[] {option, pKey, "-d", dataDirectory, "-n", "mastery"}, cfg));
+        assertThat(Keystore.list().length).isEqualTo(1);
+
+        // test 3: -d -n first
+        key = ECKeyFac.inst().create();
+        pKey = Hex.toHexString(key.getPrivKeyBytes());
+
+        assertEquals(
+                EXIT,
+                mockCli.call(
+                        new String[] {"-n", "mastery", "-d", dataDirectory, option, pKey}, cfg));
+        assertThat(Keystore.list().length).isEqualTo(2);
+
+        // test 4: ai middle
+        key = ECKeyFac.inst().create();
+        pKey = Hex.toHexString(key.getPrivKeyBytes());
+
+        assertEquals(
+                EXIT,
+                mockCli.call(
+                        new String[] {"-n", "mastery", option, pKey, "-d", dataDirectory}, cfg));
+        assertThat(Keystore.list().length).isEqualTo(3);
+    }
+
+    @Test
+    public void testImportPrivateKey_withSplitInput() {
+        // create account
+        ECKey key = ECKeyFac.inst().create();
+        String pKey = Hex.toHexString(key.getPrivKeyBytes());
+
+        assertThat(mockCli.call(new String[] {"-a", pKey}, cfg)).isEqualTo(ERROR);
+
+        assertThat(mockCli.call(new String[] {"-a", "imp", pKey}, cfg)).isEqualTo(ERROR);
+
+        assertThat(mockCli.call(new String[] {"--acc", "import", pKey}, cfg)).isEqualTo(ERROR);
+
+        assertThat(mockCli.call(new String[] {"-a", "import", pKey}, cfg)).isEqualTo(EXIT);
+        assertThat(Keystore.list().length).isEqualTo(1);
+
+        key = ECKeyFac.inst().create();
+        pKey = Hex.toHexString(key.getPrivKeyBytes());
+        assertThat(mockCli.call(new String[] {"--account", "import", pKey}, cfg)).isEqualTo(EXIT);
+        assertThat(Keystore.list().length).isEqualTo(2);
+    }
+
+    /**
+     * Ensures that the { <i>-a import</i>, <i>a</i>, <i>--account import</i> } arguments fail when
+     * a non-private key is supplied.
+     */
+    @Test
+    @Parameters({"-a import", "ai", "--account import"})
+    public void testImportNonPrivateKey(String option) {
+        String fakePKey = Hex.toHexString("random".getBytes());
+
+        assertThat(mockCli.call(new String[] {option, fakePKey}, cfg)).isEqualTo(ERROR);
+
+        assertThat(mockCli.call(new String[] {option, "hello"}, cfg)).isEqualTo(ERROR);
+    }
+
+    // Methods below taken from FileUtils class
+    private static boolean copyRecursively(File src, File target) {
```

Comment:
We should probably look into using Apache Commons IO (or something similar) for these functions instead of just copy pasting them everywhere...

Reply:
I created an issue for this refactoring #660